// optolith-database-schema/lib/types/_Activatable.js
/**
 * General type specifications used by multiple activatable entries.
 * @title Activatable
 */
/**
 * The definition of how the combat special ability can be used in combat.
 */
export var CombatSpecialAbilityUsageType;
(function (CombatSpecialAbilityUsageType) {
    CombatSpecialAbilityUsageType["Passive"] = "Passive";
    CombatSpecialAbilityUsageType["BasicManeuver"] = "BasicManeuver";
    CombatSpecialAbilityUsageType["SpecialManeuver"] = "SpecialManeuver";
})(CombatSpecialAbilityUsageType || (CombatSpecialAbilityUsageType = {}));
/**
 * The definition of if the combat special ability can be used when armed or
 * when unarmed.
 */
export var CombatSpecialAbilityType;
(function (CombatSpecialAbilityType) {
    CombatSpecialAbilityType["Armed"] = "Armed";
    CombatSpecialAbilityType["Unarmed"] = "Unarmed";
})(CombatSpecialAbilityType || (CombatSpecialAbilityType = {}));
/**
 * Set if a predefined different word should be used instead of the word
 * `attack` for display purposes.
 */
export var PenaltyByAttackReplacement;
(function (PenaltyByAttackReplacement) {
    PenaltyByAttackReplacement["Throw"] = "Throw";
})(PenaltyByAttackReplacement || (PenaltyByAttackReplacement = {}));
/**
 * The AE cost are per level of the enchantment. It may either be displayed
 * in a compressed way (e.g. `1 AE per level`) or in a verbose way (e.g. `1
 * AE for level I; 2 AE for level II`).
 */
export var FixedArcaneEnergyCostPerLevel;
(function (FixedArcaneEnergyCostPerLevel) {
    FixedArcaneEnergyCostPerLevel["Compressed"] = "Compressed";
    FixedArcaneEnergyCostPerLevel["Verbose"] = "Verbose";
})(FixedArcaneEnergyCostPerLevel || (FixedArcaneEnergyCostPerLevel = {}));
// "Input": {
//   "description": "A string that is used as a placeholder text for an input field.",
//   "type": "string",
//   "minLength": 1
// },
// "AdvancedSpecialAbilitiesAppend": {
//   "description": "An addition to the default advanced special abilities text. Markdown is available.",
//   "type": "string",
//   "minLength": 1
// },
// "ApValueReplacement": {
//   "description": "The AP value. Only use this if the text provides different information than `X adventure points`, e.g. for Special Ability Property Knowledge it is \"10 adventure points for the first *Property Knowledge*, 20 adventure points for the second, 40 adventure points for the third\". Markdown is available.",
//   "type": "string",
//   "minLength": 1
// },
// "ApValueAppend": {
//   "description": "An addition to the default AP value schema. Only use this if the text provides information appended to `X adventure points` and if `apValue` is not used. Markdown is available.",
//   "type": "string",
//   "minLength": 1
// },
// "PrerequisitesReplacement": {
//   "description": "Use if text cannot be generated by the app. Markdown is available.",
//   "type": "string",
//   "minLength": 1
// },
// "PrerequisitesStart": {
//   "description": "Prepends the provided string to the main prerequisites string. No effect if `prerequisites` field is used in l10n file. Markdown is available.",
//   "type": "string",
//   "minLength": 1
// },
// "PrerequisitesEnd": {
//   "description": "Appends the provided string to the main prerequisites string. No effect if `prerequisites` field is used in l10n table. Markdown is available.",
//   "type": "string",
//   "minLength": 1
// }
