// optolith-database-schema/lib/types/Culture.js
/**
 * @main Culture
 */
import { validateSchemaCreator } from "../validation/schema.js";
/**
 * The "weight" difference compared to other professions or profession variants.
 * Some professions or profession variants are simply more common (Mostly), but
 * sometimes only specific elements are used (Only).
 */
export var Weight;
(function (Weight) {
    Weight["Mostly"] = "Mostly";
    Weight["Only"] = "Only";
})(Weight || (Weight = {}));
/**
 * This defines how the list of constraints should be offset against the
 * list of all mundane professions: Either only the professions are kept
 * that intersect with the constraints (include) or only the professions
 * are kept that are different from the constraints (exclude).
 */
export var CommonProfessionConstraintsOperation;
(function (CommonProfessionConstraintsOperation) {
    CommonProfessionConstraintsOperation["Intersection"] = "Intersection";
    CommonProfessionConstraintsOperation["Difference"] = "Difference";
})(CommonProfessionConstraintsOperation || (CommonProfessionConstraintsOperation = {}));
/**
 * Some professions may be found in a culture, but are not that common.
 */
export var Rarity;
(function (Rarity) {
    Rarity["Rare"] = "Rare";
    Rarity["VeryRare"] = "VeryRare";
})(Rarity || (Rarity = {}));
/**
 * Some professions may be found in a culture, but are not that common.
 */
export var MundaneProfessionSubgroupConstraint;
(function (MundaneProfessionSubgroupConstraint) {
    MundaneProfessionSubgroupConstraint["Profane"] = "Profane";
    MundaneProfessionSubgroupConstraint["Fighter"] = "Fighter";
    MundaneProfessionSubgroupConstraint["Religious"] = "Religious";
})(MundaneProfessionSubgroupConstraint || (MundaneProfessionSubgroupConstraint = {}));
export const validateSchema = validateSchemaCreator(import.meta.url);
