// optolith-database-schema/lib/types/Patron.js
/**
 * @main Patron
 */
import { validateSchemaCreator } from "../validation/schema.js";
/**
 * The set operation to combine the set of all patron cultures with the
 * specified set of patron cultures: If they should intersect, the patron is
 * only part of the given cultures. If they should differ, the patron is only
 * part of the cultures that are not given.
 */
export var PatronCultureOperation;
(function (PatronCultureOperation) {
    PatronCultureOperation["Intersection"] = "Intersection";
    PatronCultureOperation["Difference"] = "Difference";
})(PatronCultureOperation || (PatronCultureOperation = {}));
export var CombatAnimalPowerType;
(function (CombatAnimalPowerType) {
    CombatAnimalPowerType["Attack"] = "Attack";
    CombatAnimalPowerType["Parry"] = "Parry";
    CombatAnimalPowerType["RangedCombat"] = "RangedCombat";
    CombatAnimalPowerType["Dodge"] = "Dodge";
    CombatAnimalPowerType["DamagePoints"] = "DamagePoints";
    CombatAnimalPowerType["Protection"] = "Protection";
})(CombatAnimalPowerType || (CombatAnimalPowerType = {}));
export const validateSchema = validateSchemaCreator(import.meta.url);
